upstash_redis/client.py

```diff
@@ -33,7 +52,7 @@ def __init__(
 
         self._allow_telemetry = allow_telemetry
 
-        self._rest_encoding = rest_encoding
+        self._rest_encoding: Union[Literal["base64"], None] = rest_encoding
```

Comment:
minor: do we need this change?

Reply:
For some reason literal type turns into str and it gives an error in vscode, thats why

<img width="406" alt="image" src="https://github.com/upstash/redis-python/assets/18082091/e6114773-22b6-4e39-9058-852e2c1702ba">
<img width="409" alt="image" src="https://github.com/upstash/redis-python/assets/18082091/b1533e5f-6954-4959-a589-b90c26cabdef">